build_settings.cmake

```diff
@@ -47,6 +47,11 @@ if (VESPA_USE_SANITIZER)
     set(C_WARN_OPTS "${C_WARN_OPTS} -Wno-tsan")
   endif()
 endif()
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 14.0)
```

Comment:
Compiler bug ?

Reply:
The symptom is error messages like:
```
In file included from /home/tegge/git/vespa/eval/src/vespa/eval/eval/test/eval_spec.h:5,
                 from /home/tegge/git/vespa/eval/src/vespa/eval/eval/test/eval_spec.cpp:3:
In member function ‘bool vespalib::small_string<StackSize>::isAllocated() const [with unsigned int StackSize = 48]’,
    inlined from ‘vespalib::small_string<StackSize>::~small_string() [with unsigned int StackSize = 48]’ at /home/tegge/git/vespa/vespalib/src/vespa/vespalib/stllike/string.h:214:41,
    inlined from ‘void vespalib::eval::test::EvalSpec::add_function_call_cases()’ at /home/tegge/git/vespa/eval/src/vespa/eval/eval/test/eval_spec.cpp:168:19:
/home/tegge/git/vespa/vespalib/src/vespa/vespalib/stllike/string.h:581:62: error: ‘<unnamed>[0].vespalib::small_string<48>::_buf’ may be used uninitialized [-Werror=maybe-uninitialized]
  581 |     [[nodiscard]] bool isAllocated() const noexcept { return _buf != _stack; }
      |                                                              ^~~~
/home/tegge/git/vespa/eval/src/vespa/eval/eval/test/eval_spec.cpp: In member function ‘void vespalib::eval::test::EvalSpec::add_function_call_cases()’:
/home/tegge/git/vespa/eval/src/vespa/eval/eval/test/eval_spec.cpp:168:19: note: ‘<anonymous>’ declared here
  168 |     add_expression({"a"}, "isNan(a)")
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
```

When compiling with `make -k`, there are 17 such errors, all of the compliants are about _buf in vespalib::string not being initialized. The first 16 are via calls to add_expression as shown above, while the last is via 
```
In file included from /home/tegge/git/vespa/vespalib/src/vespa/vespalib/util/time.h:6,
                 from /home/tegge/git/vespa/vespalib/src/vespa/vespalib/util/count_down_latch.h:5,
                 from /home/tegge/git/vespa/vespalib/src/vespa/vespalib/testkit/test_hook.h:5,
                 from /home/tegge/git/vespa/vespalib/src/vespa/vespalib/testkit/test_kit.h:10,
                 from /home/tegge/git/vespa/vespalib/src/tests/state_server/state_server_test.cpp:3:
In member function ‘bool vespalib::small_string<StackSize>::isAllocated() const [with unsigned int StackSize = 48]’,
    inlined from ‘vespalib::small_string<StackSize>::~small_string() [with unsigned int StackSize = 48]’ at /home/tegge/git/vespa/vespalib/src/vespa/vespalib/stllike/string.h:214:41,
    inlined from ‘void {anonymous}::TestKitHook266::Test<F1>::test_entry_point() [with F1 = vespalib::JsonHandlerRepo]’ at /home/tegge/git/vespa/vespalib/src/tests/state_server/state_server_test.cpp:270:5:
/home/tegge/git/vespa/vespalib/src/vespa/vespalib/stllike/string.h:581:62: error: ‘<unnamed>[0].vespalib::small_string<48>::_buf’ may be used uninitialized [-Werror=maybe-uninitialized]
  581 |     [[nodiscard]] bool isAllocated() const noexcept { return _buf != _stack; }
      |                                                              ^~~~
In file included from /home/tegge/git/vespa/vespalib/src/vespa/vespalib/testkit/test_kit.h:7:
/home/tegge/git/vespa/vespalib/src/tests/state_server/state_server_test.cpp: In member function ‘void {anonymous}::TestKitHook266::Test<F1>::test_entry_point() [with F1 = vespalib::JsonHandlerRepo]’:
/home/tegge/git/vespa/vespalib/src/tests/state_server/state_server_test.cpp:270:22: note: ‘<anonymous>’ declared here
  270 |     EXPECT_TRUE(std::vector<vespalib::string>({"/health"}) == f1.get_root_resources());
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tegge/git/vespa/vespalib/src/vespa/vespalib/testkit/test_macros.h:53:48: note: in definition of macro ‘EXPECT_TRUE’
   53 | #define EXPECT_TRUE(rc) TEST_MASTER.check(bool(rc), __FILE__, __LINE__, TEST_STR(rc), false)
      |                                                ^~
cc1plus: all warnings being treated as errors
```

My guess is that the compiler gets confused and tries to blame vespalib::string constructors.